src/script.ts

```diff
@@ -110,45 +122,190 @@ export const runScript = async (
   return k6
 }
 
-const enhanceScript = async (scriptPath: string) => {
-  const groupSnippet = await getJsSnippet('group_snippet.js')
-  const checksSnippet = await getJsSnippet('checks_snippet.js')
-  const scriptContent = await readFile(scriptPath, { encoding: 'utf-8' })
-  const scriptLines = scriptContent.split('\n')
-  const httpImportIndex = scriptLines.findIndex((line) =>
-    line.includes('k6/http')
+const parseScript = (input: string) => {
+  return parse(input, {
+    loc: false,
+    range: false,
+  })
+}
+
+/**
+ * It's theoretically possible that the user has imported the `k6/execution` module with the
+ * same alias as our shim. In that case we need to remove the conflicting import to avoid a
+ * syntax error. If they imported it using a different alias then there's no harm in keeping
+ * it around.
+ */
+function isConflictingExecutionImport(node: ts.Node) {
+  return (
+    node.type === NodeType.ImportDeclaration &&
+    node.source.value === 'k6/execution' &&
+    node.specifiers.some(
+      (specifier) =>
+        specifier.type === NodeType.ImportDefaultSpecifier &&
+        specifier.local.name === 'execution'
+    )
   )
-  const handleSummaryIndex = scriptLines.findIndex(
-    (line) =>
-      // NOTE: if the custom handle summary is commented out we can still insert our snippet
-      // this check should be improved
-      line.includes('export function handleSummary(') && !line.includes('//')
+}
+
+interface EnhanceScriptOptions {
+  script: string
+  shims: {
+    group: string
+    checks: string
+  }
+}
+
+export const enhanceScript = async ({
+  script,
+  shims,
+}: EnhanceScriptOptions) => {
+  const [groupAst, checksAst, scriptAst] = await Promise.all([
+    parseScript(shims.group),
+    parseScript(shims.checks),
+    parseScript(script),
+  ])
+
+  if (groupAst === undefined) {
+    throw new Error('Failed to parse group snippet')
+  }
+
+  if (checksAst === undefined) {
+    throw new Error('Failed to parse checks snippet')
+  }
+
+  if (scriptAst === undefined) {
+    throw new Error('Failed to parse script content')
+  }
```

Comment:
Not sure what we should do if the user imports an invalid script file. Right now I just throw, but maybe I should return the unaltered "script" and let k6 fail when trying to parse it?